src/renderer/store/modules/settings.js

```diff
@@ -428,6 +436,24 @@ const customActions = {
           commit(defaultMutationId(_id), value)
         }
       }
+
+      for (const setting of userSettings) {
+        const [_id, value] = setting
+        loadSetting(_id, value)
+      }
+
+      // Apply existing values of outdated setting variables in the DB to their newer equivalents,
+      // then delete those older settings
+      for (const outdatedSetting of Object.keys(outdatedSettings)) {
+        const outdatedSettingInDB = userSettings.find((setting) => setting[0] === outdatedSetting)
+        if (!outdatedSettingInDB) {
+          return
+        }
+        const newSetting = outdatedSettings[outdatedSetting]
+        const oldValue = outdatedSettingInDB[1]
+        loadSetting(newSetting, oldValue)
+        await DBSettingHandlers.delete(outdatedSetting)
```

Comment:
@absidue  Your suggestion has been implemented. Only last question is if we want to comment this line out for one release to allow new release toe-dippers from being disrupted.

Reply:
I like this more but it's still not quite what I had in mind.

My idea was that the fixing of the settings would happen inside the datastore base handler settings class (`src/datastores/handlers/base.js`) in the relevant query functions e.g. `find()`, `_findAppReadyRelatedSettings()`, etc (you'll have to check which ones are actually affected). That way it would be the only place that would have to have any code for renamed settings, the rest of FreeTube like the store, would not have to care that settings were ever renamed, as it's handled in one place at the lowest possible level.

Reply:
Want to make sure I understand what you mean - have application logic in `src/datastores/handlers/base.js` that queries for a specific set of settings before `find()`, uses each one's value to supplant their new equivalent's, and then deletes them? We don't have any other app logic there, so I worry that might be a bit out of place if I'm reading you right.

Reply:
@absidue Apologies, just boosting this q in case you didn't see.

Reply:
Personally I see these changes as database migrations, which I would prefer to keep as close to the actual databases as possible. It also keeps the code a lot simpler, as only place that has to care about the settings having another name the `Settings.find()` method in the `src/datastores/handlers/base.js` file, instead of your current approach that spreads it across many files and also requires IPC calls to make the changes, instead of being able to directly modify the database.

You wouldn't need any extra find calls. just do the normal find all call on the database, modify it if necessary, make the database modifications if necessary and then return the response, that can all happen inside the `Settings.find()` method in the `src/datastores/handlers/base.js` file.